### DISCUSSION TITLE
Different init_range

### DISCUSSION THREAD

**Discussion by itmamdiyar26** [ahmedfgad/GeneticAlgorithmPython#53] [Q&A] [answered]
Mr Gad,
I have a problem I hope you can help me with this problem
I have 3 inputs, namely x,y,z and each input has a different starting and ending range
can this be done?
Thanks for your attention

**Comment by ahmedfgad** [comment, marked as answer]
Hi @itmamdiyar26,

This feature is already supported in PyGAD using the `gene_space` parameter.

Here is an example to have a range of discrete (integers only) values.

```python
gene_space=[range(-5, 10), range(50, 70), range(3, 18)]
```

Read this for more information: https://pygad.readthedocs.io/en/latest/README_pygad_ReadTheDocs.html#more-about-the-gene-space-parameter

Please ask if you have any questions.
